web/src/server/api/routers/annotationQueues.ts

```diff
@@ -0,0 +1,532 @@
+          resourceId: queue.id,
+          action: "create",
+          after: queue,
+        });
+
+        return queue;
+      } catch (error) {
+        logger.error(error);
+        if (error instanceof TRPCError) {
+          throw error;
+        }
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: "Creating annotation queue failed.",
+        });
+      }
+    }),
+  update: protectedProjectProcedure
+    .input(
+      CreateQueueData.extend({
+        projectId: z.string(),
+        queueId: z.string(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      try {
+        throwIfNoEntitlement({
+          entitlement: "annotation-queues",
+          projectId: input.projectId,
+          sessionUser: ctx.session.user,
+        });
+
+        throwIfNoProjectAccess({
+          session: ctx.session,
+          projectId: input.projectId,
+          scope: "annotationQueues:CUD",
+        });
+
+        const queue = await ctx.prisma.annotationQueue.findFirst({
+          where: { id: input.queueId, projectId: input.projectId },
+        });
+
+        if (!queue) {
+          throw new LangfuseNotFoundError("Queue not found in project");
+        }
+
+        const existingQueue = await ctx.prisma.annotationQueue.findFirst({
+          where: {
+            projectId: input.projectId,
+            name: input.name,
+          },
+        });
+
+        if (existingQueue) {
+          throw new TRPCError({
+            code: "CONFLICT",
+            message: "A queue with this name already exists in the project",
+          });
+        }
+
+        const updatedQueue = await ctx.prisma.annotationQueue.update({
+          where: { id: input.queueId, projectId: input.projectId },
+          data: {
+            name: input.name,
+            description: input.description,
+            scoreConfigs: input.scoreConfigs,
+          },
+        });
+
+        await auditLog({
+          session: ctx.session,
+          resourceType: "annotationQueue",
+          resourceId: queue.id,
+          action: "update",
+          before: queue,
+          after: updatedQueue,
+        });
+
+        return updatedQueue;
+      } catch (error) {
+        logger.error(error);
+        if (error instanceof TRPCError) {
+          throw error;
+        }
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: "Updating annotation queue failed.",
+        });
+      }
+    }),
+  delete: protectedProjectProcedure
+    .input(z.object({ queueId: z.string(), projectId: z.string() }))
+    .mutation(async ({ input, ctx }) => {
+      try {
+        throwIfNoEntitlement({
+          entitlement: "annotation-queues",
+          projectId: input.projectId,
+          sessionUser: ctx.session.user,
+        });
+
+        throwIfNoProjectAccess({
+          session: ctx.session,
+          projectId: input.projectId,
+          scope: "annotationQueues:CUD",
```

Comment:
maybe we should change the seeder users to get the team plan so that we can test this feature after running pnpm run dx